src/emailpro/external-contact/add/emailpro-external-contact-add.controller.js

```diff
@@ -0,0 +1,110 @@
+angular
+    .module("Module.emailpro.controllers")
+    .controller("EmailProAddExternalContactCtrl", class EmailProAddExternalContactCtrl {
+        constructor ($scope, $stateParams, EmailPro, EmailProExternalContacts, Alerter, translator) {
+            this.$scope = $scope;
+            this.$stateParams = $stateParams;
+            this.EmailPro = EmailPro;
+            this.EmailProExternalContacts = EmailProExternalContacts;
+            this.Alerter = Alerter;
+            this.translator = translator;
+        }
+
+        $onInit () {
+            this.model = {
+                hiddenFromGAL: false
+            };
+
+            this.isLoading = false;
+            this.emailValidationRegex = "^[a-zA-Z0-9]+([\.+\-\w][a-zA-Z0-9]+)*@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$";
+
+            this.EmailPro
+                .getSelected()
+                .then((exchange) => {
+                    if (exchange.serverDiagnostic.version === 14 && exchange.offer === this.EmailPro.accountTypeProvider) {
+                        this.EmailProExternalContacts.getContactOptions(this.$stateParams.organization, this.$stateParams.productId)
+                            .then((data) => {
+                                this.availableMainDomains = data;
+                                this.attachOrganization2010 = this.availableMainDomains[0];
+                            })
+                            .catch((failure) => {
+                                this.$scope.resetAction();
+                                this.$scope.setMessage(this.translator.tr("exchange_tab_EXTERNAL_CONTACTS_configuration_contact_add_fail"), failure.data);
+                            });
+                    }
+                });
+
+            this.retrievingAlreadyTakenEmails();
+
+            this.$scope.submitting = () => this.submitting();
+        }
+
+        retrievingAlreadyTakenEmails () {
+            this.isLoading = true;
+
+            return this.EmailProExternalContacts
+                .retrievingContacts(this.$stateParams.productId)
+                .then((data) => {
+                    this.alreadyTakenEmails = data.list.results.map((contact) => contact.externalEmailAddress);
+                })
+                .catch((error) => {
+                    this.Alerter.alertFromSWS(this.translator.tr("exchange_tab_EXTERNAL_CONTACTS_configuration_contact_add_fail"), error, this.$scope.alerts.dashboard);
+                    this.$scope.resetAction();
+                })
+                .finally(() => {
+                    this.isLoading = false;
+                });
+        }
+
+        submitting () {
+            this.$scope.resetAction();
+
+            if (!_.isEmpty(this.attachOrganization2010)) {
+                this.model.organization2010 = this.attachOrganization2010.name;
```

Comment:
Can it be simplify ? (directly play with `this.model.organization2010` to avoid copying it when you submit the form)